src/simtools/model/model_parameter.py

```diff
@@ -446,40 +438,67 @@ def export_parameter_file(self, par_name, file_path):
         shutil.copy(file_path, self.config_file_directory)
 
-    def export_model_files(self):
-        """Export the model files into the config file directory."""
+    def export_model_files(self, destination_path=None, update_if_necessary=False):
+        """
+        Export the model files into the config file directory.
+
+        Parameters
+        ----------
+        destination_path: str
+            Path to the directory where the model files should be exported.
+            If None, the config file directory is used.
+        update_if_necessary: bool
+            If True, the model files are only exported if they are not up to date.
+        """
+        if self._is_exported_model_files_up_to_date and update_if_necessary:
+            self._logger.debug(
+                f"Model files for {self.name} are already exported to {self.config_file_directory}"
+            )
+            return
         # Removing parameter files added manually (which are not in DB)
         pars_from_db = copy(self.parameters)
         if self._added_parameter_files is not None:
             for par in self._added_parameter_files:
                 pars_from_db.pop(par)
 
-        self.db.export_model_files(parameters=pars_from_db, dest=self.config_file_directory)
+        self.db.export_model_files(
+            parameters=pars_from_db,
+            dest=destination_path or self.config_file_directory,
+        )
         self._is_exported_model_files_up_to_date = True
 
-    def export_config_file(self):
-        """Export the config file used by sim_telarray."""
-        if not self._is_exported_model_files_up_to_date:
-            self.export_model_files()
+    def write_sim_telarray_config_file(self, add_model=None):
+        """
+        Write the sim_telarray configuration file.
+
+        Parameters
+        ----------
+        add_model: TelescopeModel or SiteModel
+            Model object for additional parameter to be written to the config file.
+        """
+        self.export_model_files(update_if_necessary=True)
+
+        self.parameters.update(self._simulation_config_parameters.get("simtel", {}))
+        if add_model:
+            self.parameters.update(add_model.parameters)
+            add_model.export_model_files(self.config_file_directory, update_if_necessary=True)
 
         self._load_simtel_config_writer()
         self.simtel_config_writer.write_telescope_config_file(
             config_file_path=self.config_file_path,
             parameters=self.parameters,
-            config_parameters=self._simulation_config_parameters["sim_telarray"],
         )
 
     @property
     def config_file_directory(self):
         """Directory for configure files. Configure, if necessary."""
         if self._config_file_directory is None:
             self._set_config_file_directory_and_name()
-        return self._config_file_directory
+        return self._config_file_directory or self._set_config_file_directory_and_name()
```

Comment:
Wouldn't this make the code above it checking if `None` redundant? I suggest applying the same lazy init to the config_file_path while we are at it.

Reply:
Yes. Fixed it.